Building an API.md

```diff
@@ -0,0 +1,53 @@
+# Intro
+
+Building an API is different than building a regular HTML application. There are
+no HTML views to worry about, but instead you should take extra care about status responses, response format and especially documentation since your application will be used
+by other developers.
+
+## Creating an application
+
+To generate a Rails application with only the elements necessary for an API
+application use the `--api` flag e.g.
+`rails new my_application --api`
+
+We write our APIs in JSON format, more specifically using the [JSON API standard](http://jsonapi.org/).
+This allows those who consume our APIs to know what to expect regarding the
+document structure.
+
+## Gems
+
+We use a few gems that make our lives easier when writing APIs:
+* [json_api_responders](https://github.com/infinum/json_api_responders)
+* Simplifies the code in controllers, no need to write `if @object.save` etc. to
+display errors.
+* [jsonapi-rails](https://github.com/jsonapi-rb/jsonapi-rails)
+* Takes care of serialization/deserialization of Ruby objects into/out of JSON
+objects, all according to the JSON API standard, so you don't have to think about
+those details.
+* [rspec](https://github.com/rspec/rspec-rails)
+* Testing is always important when building software, but doubly so when writing
+an API since we can use tests to generate documentation.
+* [dox](https://github.com/infinum/dox)
+* Generates documentation from rspec tests in api blueprint format.
+
+## Documentation
+
+You can generate documentation from your controller or request specs, and it's very
+important that you cover all the possible cases and return codes e.g. a valid response,
+response when there are errors and any other outcome of the action. Also document any filtering parameters and enumerations that may be used to refine the results of the API call.
```

Comment:
Make a list of what should be in the documentation, so it's more noticeable. 

Add also supported sorting parameters (as requesting sort with unsupported param should return 400), how the authentication is done, pagination parameters (with default and max page size). 

Re hosting documentation - it's best to put the documentation regeneration and push in the  CI process (after successful build), so the documentation is always up to date.